mmv1/third_party/terraform/services/composer/resource_composer_environment.go.erb

```diff
@@ -900,32 +900,2653 @@ func ResourceComposerEnvironment() *schema.Resource {
 				Description: `All of labels (key/value pairs) present on the resource in GCP, including the labels configured through Terraform, other clients and services.`,
 				Elem:        &schema.Schema{Type: schema.TypeString},
 			},
+
+			"storage_config": {
+				Type:         schema.TypeList,
+				Optional:     true,
+				Computed:     true,
+				MaxItems:     1,
+				Description:  `Configuration options for storage used by Composeer environment.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"bucket": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							Description: `Optional. Name of an existing Cloud Storage bucket to be used by the environment.`,
+						},
+					},
+				},
+			},
 		},
 		UseJSONNumber: true,
 	}
 }
 
-func resourceComposerEnvironmentCreate(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*transport_tpg.Config)
-	userAgent, err :=  tpgresource.GenerateUserAgentString(d, config.UserAgent)
-	if err != nil {
-		return err
-	}
+func resourceComposerEnvironmentCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	userAgent, err :=  tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+
+	envName, err := resourceComposerEnvironmentName(d, config)
+	if err != nil {
+		return err
+	}
+
+	transformedConfig, err := expandComposerEnvironmentConfig(d.Get("config"), d, config)
+	if err != nil {
+		return err
+	}
+
+	transformedStorageConfig, err := expandComposerStorageConfig(d.Get("storage_config"), d, config)
+	if err != nil {
+		return err<% autogen_exception -%>
+		package composer
+
+		import (
```

Comment:
This seems to have gotten mixed up pretty badly- we're cloning most of the file in the middle of a function right now.